From: Chuanhong Guo <gch981213@gmail.com>
Date: Sat, 9 Apr 2022 17:13:47 +0800
Subject: arm: mediatek: select arch timer for mt7629
Git-commit: d66aea197d534e23d4989eb72fca9c0c114b97c9
Patch-mainline: v5.18 or v5.18-rc8 (next release)
References: git-fixes

This chip has an armv7 arch timer according to the dts. Select it in
Kconfig to enforce the support for it.
Otherwise the system time is just completely wrong if user forget to
enable ARM_ARCH_TIMER in kernel config.

Fixes: a43379dddf1b ("arm: mediatek: add MT7629 smp bring up code")
Signed-off-by: Chuanhong Guo <gch981213@gmail.com>
Link: https://lore.kernel.org/r/20220409091347.2473449-1-gch981213@gmail.com
Signed-off-by: Matthias Brugger <matthias.bgg@gmail.com>
Acked-by: Ivan T. Ivanov <iivanov@suse.de>
---
 arch/arm/mach-mediatek/Kconfig |    1 +
 1 file changed, 1 insertion(+)

--- a/arch/arm/mach-mediatek/Kconfig
+++ b/arch/arm/mach-mediatek/Kconfig
@@ -30,6 +30,7 @@ config MACH_MT7623
 config MACH_MT7629
 	bool "MediaTek MT7629 SoCs support"
 	default ARCH_MEDIATEK
+	select HAVE_ARM_ARCH_TIMER
 
 config MACH_MT8127
 	bool "MediaTek MT8127 SoCs support"
